unidist/core/backends/ray/backend.py

```diff
@@ -137,6 +137,26 @@ def num_cpus():
         """
         return int(ray.cluster_resources()["CPU"])
 
+    @staticmethod
+    def cluster_resources():
+        """
+        Get resources of Ray cluster.
+
+        Returns
+        -------
+        dict
+            Dictionary with cluster nodes info in the form '{node_ip0: {CPU: x0},
+            node_ip1: {CPU: x1}, ..}'.
+        """
+        node_resources = {
+            node_info["NodeManagerAddress"]: {
```

Comment:
Do `NodeManagerAddress`-s match each node in the cluster? Should we use node_info["Resources"]["node:IP"] instead?

Reply:
Yes, `NodeManagerAddress` matches each node in the cluster. For example, for 2-nodes cluster the output of `ray.nodes()`:
```python
[{'NodeID': 'd39331290831201ad3c62b8e9419b4551fb20cd68ae4df196ddec233',
  'Alive': True,
  'NodeManagerAddress': '172.31.25.69',
  'NodeManagerHostname': 'ip-172-31-25-69',
  'NodeManagerPort': 43333,
  'ObjectManagerPort': 44771,
  'ObjectStoreSocketName': '/tmp/ray/session_2022-03-21_08-34-56_949138_4880/sockets/plasma_store',
  'RayletSocketName': '/tmp/ray/session_2022-03-21_08-34-56_949138_4880/sockets/raylet',
  'MetricsExportPort': 56201,
  'alive': True,
  'Resources': {'node:172.31.25.69': 1.0,
   'memory': 6445255067.0,
   'object_store_memory': 3222627532.0,
   'CPU': 4.0}},
 {'NodeID': '94f384f3b0b3f4a214d5737537b63128d26654df1d7286c2f74466bc',
  'Alive': True,
  'NodeManagerAddress': '172.31.24.172',
  'NodeManagerHostname': 'ip-172-31-24-172',
  'NodeManagerPort': 34849,
  'ObjectManagerPort': 34331,
  'ObjectStoreSocketName': '/tmp/ray/session_2022-03-21_08-34-56_949138_4880/sockets/plasma_store',
  'RayletSocketName': '/tmp/ray/session_2022-03-21_08-34-56_949138_4880/sockets/raylet',
  'MetricsExportPort': 58259,
  'alive': True,
  'Resources': {'CPU': 4.0,
   'object_store_memory': 4371087360.0,
   'node:172.31.24.172': 1.0,
   'memory': 10199203840.0}}]
```

Using of `NodeManagerAddress` looks more native.

Reply:
ok